server/src/main/java/org/elasticsearch/common/geo/SpatialStrategy.java

```diff
@@ -54,6 +54,12 @@ public static SpatialStrategy fromString(String strategyName) {
                 return strategy;
             }
         }
-        return null;
+        throw new IllegalArgumentException("Unknown strategy [" + strategyName + "]");
+    }
```

Comment:
This ended up catching a pre-existing bug in GeoShapeQueryTests, which was asking for `TERM` instead of `term` as a strategy, and therefore ending up falling back on `null` here, which was getting translated to `recursive` once the mapper had been parsed.  Silently falling back to `recursive` is a bug, I think, although on backport we will need to change this to emit a warning instead, as you can't change the strategy once it has been set and so existing mappings with badly-spelled `term` strategies will have to stay using `recursive`

Reply:
Do you think this is a breaking change?

Reply:
I'd say it was a bugfix rather than a breaking change, I think?

Reply:
I opened #63975 to emit a deprecation warning for this case in 7.x

Reply:
Existing mappings sort of fix themselves, btw, so my above comment was not correct.  Once a user has added a mapping with a bad strategy, it will serialize itself as `recursive`, so we don't need to worry about parsing mappings from previous versions.

Reply:
Yes, that is what I checked so the only change is that creating a mapping will throw an error when before it was successful.